### ISSUE TITLE
Launcher stuck at 47%

### ISSUE THREAD

**Issue by azaroma** [Nefelim4ag/League-Of-Legends#16]
Hi, I'm running Manjaro with kernel 4.4. I have tested your script with wine-staging 1.9.23 and latest wine in repos 1.9.24 with same results. The launcher downloads everything just fine but when it hits 47% and is applying patches it gets stuck, actually, after two hours more or less it gets to 52% maximum. When I close the launcher a bugsplat window comes up.  What should I check to know what's wrong?

This is the tail output

    INFO: Check: Base dir /home/azaroma/.local/share/leagueoflegends
    INFO: Check: wineprefix exist /home/azaroma/.local/share/leagueoflegends/WINE.win32/
    INFO: Last tested version: wine-1.9.24
    INFO: Your version: wine-1.9.23 (Staging)
    INFO: Check: wineprefix configured
    INFO: Check: wine can access only to wine drive C:/ + X:/ (LoL)
    ==> /home/azaroma/.local/share/leagueoflegends/lol_launcher.log <==
    fixme:heap:RtlSetHeapInformation (nil) 1 (nil) 0 stub
    fixme:ntdll:EtwRegisterTraceGuidsW (0x10144ff0, 0x11fec2e0, {3dada31d-19ef-4dc1-b345-037927193422}, 1, 0x11fc0360, (null), (null), 0x11fec2f8): stub
    fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {00000000-0000-0000-0000-000000000000}
    fixme:gdi:GdiInitializeLanguagePack stub
    fixme:ver:GetCurrentPackageId (0x283e9d8 (nil)): stub
    fixme:ntdll:NtLockFile I/O completion on lock not implemented yet
    fixme:module:load_library unsupported flag(s) used (flags: 0x00000800)
    err:ole:CoInitializeEx Attempt to change threading model of this apartment from
    multi-threaded to apartment threaded
    fixme:cryptnet:verify_cert_revocation_from_aia_ext OCSP URL = L"http://ocsp2.globalsign.com/cloudsslsha2g3"
    fixme:system:SystemParametersInfoW Unknown action: 116
    
    ==> /home/azaroma/.local/share/leagueoflegends/lol_client.log <==
    
What other information can i provide?

**Comment by Nefelim4ag** [comment]
I don't know, but it's can be releated to https://github.com/Nefelim4ag/League-Of-Legends/issues/9

**Comment by azaroma** [comment]
Yes, it could be, but I have run `leagueoflegends regen_wine_prefix` and then `leagueoflegends launcher` again and had the same results. Its strange that the first part of the update gets recognized because the download starts at 33% now, then the launcher downloads like 3 ~ 4 Gb and gets to 47% where it gets really slow.

**Comment by azaroma** [comment]
This is tail's output when I closed the launcher before reaching 47%

    ==> /home/azaroma/.local/share/leagueoflegends/lol_launcher.log <==
    (  ERROR)[08:57:05.892] RADS::User::PandoManager::start: StartPMB failed, returned error 6
    (  ERROR)[09:31:51.398] RADS::UserKernel::NamedPipe::ConnectionThread::ThreadProc: The handle that was passed to the API has been either invalidated or closed.
    (  ERROR)[09:31:51.398] RADS::UserKernel::NamedPipe::ConnectionThread::ThreadProc: ConnectNamedPipe failed.
    (  ERROR)[09:31:51.404] RADS::Common::HTTPConnection::GetFile: perform request failed with code 7
    (  ERROR)[09:31:51.404] RADS::Common::HTTPConnection::GetFile: Failed on /rest/getToken?format=xml&ClientID=PMB_Client
    (  ERROR)[09:31:51.408] RADS::Common::HTTPConnection::GetFile: perform request failed with code 7
    (  ERROR)[09:31:51.408] RADS::Common::HTTPConnection::GetFile: Failed on /rest/getToken?format=xml&ClientID=PMB_Client
    (  ERROR)[09:31:51.409] RADS::Common::HTTPConnection::GetFile: perform request failed with code 7
    (  ERROR)[09:31:51.409] RADS::Common::HTTPConnection::GetFile: Failed on /rest/shutdown?format=xml&netToken=
    fixme:ver:GetCurrentPackageId (0x33ef04 (nil)): stub

As you can see this line is missing:

    fixme:cryptnet:verify_cert_revocation_from_aia_ext OCSP URL = L"http://ocsp2.globalsign.com/cloudsslsha2g3"

Could this be the issue?

**Comment by egeskow** [comment]
It takes ~24 hours to fresh install on my machine. Each percent from 47 is really long running. I have used winehq-staging from ppa

**Comment by rlrs** [comment]
The main reason it takes so long to do a fresh install, is that the patcher creates tens or hundreds of thousands of temporary files in the RADS/temp folder. You can create a tmpfs mount at RADS/temp instead of the folder and it'll speed up by 10x.

**Comment by azaroma** [comment]
@rlrs Only the `RADS/temp` folder should be mounted as tmpfs? It would make sense to do it as well to the folder where the patches are beign applied, since is this part of the process that takes decades

**Comment by rlrs** [comment]
Well I just tested it now and got a bugsplat at 54% when doing a fresh install :( It does seem to be faster with the tmpfs folder, but maybe not? Weird. 
You shouldn't make any other folder tmpfs since files there will disappear when your pc turns off.

**Comment by azaroma** [comment]
@rlrs It eventually worked with your method but after several tries. I didn't noticed what happened, it suddenly advanced a few percentage points and it went on from there, so can't really pin what I did

**Comment by ardarandir** [comment]
i would suspect your wine setup is faulty, perhaps missing dlls. did the script set up mono and gecko properly? are you using a recent winetricks, to make sure it can resolve the download paths correctly?

see #20 for reference. a total delete and rerun of the script would be advisable anyways. check terminal output for ERR or ERROR message. focus on 404's and ignore fixme's, as they are unlikekly to be fatal.
